src/uct/ib/mlx5/gga/gga_mlx5.c

```diff
@@ -118,41 +141,105 @@ uct_gga_mlx5_rkey_unpack(uct_component_t *component, const void *rkey_buffer,
         return UCS_ERR_NO_MEMORY;
     }
 
-    rkey_handle->packed_mkey    = *mkey;
-    /* memh and rkey_ob will be initialized on demand since MD
-     * (required for memh import) is not available at this point */
-    rkey_handle->memh           = NULL;
-    rkey_handle->md             = NULL;
-    rkey_handle->rkey_ob.rkey   = UCT_INVALID_RKEY;
-    rkey_handle->rkey_ob.handle = NULL;
-    rkey_handle->rkey_ob.type   = NULL;
+    ucs_list_head_init(&rkey_handle->mds);
+    rkey_handle->md          = NULL;
+    rkey_handle->rkey        = UCT_INVALID_RKEY;
+    rkey_handle->packed_mkey = *mkey;
 
     *rkey_p   = (uintptr_t)rkey_handle;
     *handle_p = NULL;
     return UCS_OK;
 }
 
+static ucs_status_t
+uct_gga_mlx5_md_put_rkey(uct_gga_mlx5_md_t *md,
+                         const uct_gga_mlx5_rkey_hash_key_t key,
```

Comment:
minor: since key is primitive `const` is redundant

Reply:
it's more for readability and clean API + `uct_gga_mlx5_rkey_hash_key_t ` might be changed in future, so this place won't require changes